File: WMS-Core/Models/ItemModel.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace WMS_Core.Models
{
    [Table("items")]
    public class ItemModel
    {
        [Column("id")] public int Id { get; set; }
        [Column("name")] public string? Name { get; set; }
        [Column("type")] public int TypeId { get; set; }
        [ForeignKey("TypeId")] public ItemTypeModel? ItemTypeModel { get; set; }
        [Column("weight")] public int Weight { get; set; }
        [Column("date", TypeName = "Date")] public DateTime ShelfLife { get; set; }
        [Column("image", TypeName = "Bytea")] public byte[] Image { get; set; }
        [Column("about")] public string? About { get; set; }
        [Column("warehouse")]
        [ForeignKey("warehouse")]
        public int WarehouseId { get; set; }
        [Column("price")] public int Price { get; set; }
        [Column("quantity")] public int Quantity { get; set; }
        [Column("new_price")] public int NewPrice { get; set; }
        [Column("moving")] public bool Moving { get; set; }
        
        //
    }
}

File: Shop/ClientApp/src/app/models/ItemModel.ts


export class ItemModel {
  public id?: number;
  public name?: string;
  public type?: number;
  public weight?: number;
  public shelfLife?: Date;
  public image?: string;
  public about?: string;
  public warehouse?: number;
  public price?: number;
  public quantity?: number;
  public new_price?: number;
  constructor(
    source: any
    /*
     id?: number,
    name?: string,
    type?: number,
    weight?: number,
    shelfLife?: Date,
    image?: string,
    about?: string,
    warehouse?: number,
    price?: number,
    quantity?: number,
    new_price?: number
    */
  ) {
    this.id = source.id;
    this.name = source.name;
    this.type = source.type;
    this.weight = source.weight;
    this.shelfLife = source.shelfLife;
    this.image = source.image;
    this.about = source.about;
    this.warehouse = source.warehouse;
    this.price = source.price;
    this.quantity = source.quantity;
    this.new_price = source.new_price;
  }
  toJSON() {
    return {
      id: this.id,
      name: this.name,
      type: this.type,
      weight: this.weight,
      shelfLife: this.shelfLife,
      image: this.image,
      about: this.about,
      warehouse: this.warehouse,
      price: this.price,
      quantity: this.quantity,
      new_price: this.new_price
    }
  }
}
